src/scratch/UCERF3/enumTreeBranches/ScalingRelationships.java

```diff
@@ -252,12 +252,69 @@ public double getArea(double mag, double origWidth) {
 			return map;
 		}
 
-	};
+	},
+	
+	TMG_SUB_2017("Thingbaijam et al.(2017) Subduction", "TMG_SUB_2017") {
+		// units of the input dimensions are in m or m^2 
+		public double getAveSlip(double area, double length, double origWidth) {
+			tmg_sub_magArea.setRake(90.0d); // interface
```

Comment:
I had a look at how other scalingrelationships handle this, and none of them store state like this. In fact it seems they make sure they're threadsafe. I think I did accept this in my review for `TMG2017SubMagAreaRel`, but now I think that was a mistake. `TMG2017SubMagAreaRel` should not have state.

Reply:
Yes, I see what you're saying. I think the problem stems from the base class...

```
package org.opensha.commons.calc.magScalingRelations;
....
public abstract class MagScalingRelationship 
```

as this is where the setRake() method is defined. Hard to know what the impact would be of changing this API. A couple of suggestions...

 - make the setRake method protected or even private.
 - create a constructor that passes in rake.

all that said, the way it's being used now for TMG shouldn't cause problems, thought that's more good luck than anything :)

Reply:
I suggest we raise a separate ticket for this one.

Reply:
here's the ticket https://github.com/GNS-Science/opensha-ucerf3/issues/25